ultralytics/utils/__init__.py

```diff
@@ -347,8 +347,16 @@ def set_logging(name="LOGGING_NAME", verbose=True):
     """
     level = logging.INFO if verbose and RANK in {-1, 0} else logging.ERROR  # rank in world for Multi-GPU trainings
 
+    class BaseFormatter(logging.Formatter):
+        def format(self, record):
+            if record.levelno == logging.WARNING:
+                record.msg = colorstr("yellow", "bold", record.msg)
+            if record.levelno == logging.DEBUG:
+                record.msg = colorstr("magenta", "bold", record.msg)
+            return super().format(record)
+
     # Configure the console (stdout) encoding to UTF-8, with checks for compatibility
-    formatter = logging.Formatter("%(message)s")  # Default formatter
+    formatter = BaseFormatter()  # Default formatter
     if WINDOWS and hasattr(sys.stdout, "encoding") and sys.stdout.encoding != "utf-8":
 
         class CustomFormatter(logging.Formatter):
```

Comment:
also since we have a `CustomFormatter` for windows that inherited from `logging.Formatter` perhaps we need to update here to:
```python
class CustomFormatter(BaseFormatter):
```